### ISSUE TITLE
Hammerting (760650)

### ISSUE THREAD

**Issue by Sebastiangperez** [ValveSoftware/Proton#4347]
# Compatibility Report
- Name of the game with compatibility issues: Hammering
- Steam AppID of the game: 760650

## System Information
- GPU: GT 980
- Driver/LLVM version: Nvidia 440.44
- Kernel version: 5.0.16
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: 5.13

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-760650.log](https://github.com/ValveSoftware/Proton/files/5482905/steam-760650.log)

## Symptoms 
The game starts the first time but when you exit the game and try to play the second time keeps running in the background , i change the mode from borderless fullscreen to window mode. 

## Reproduction
Press the play button.

**Comment by Sebastiangperez** [comment]
I wanna add new stuff here : 
The game works but takes a little bit to start, like 10 o 20 seconds. 
Works on Proton 5.13 , but with the latest update of Proton 5.0, it requires Vulkan 1.1 and asks that i have to update my Nvidia Drivers.

**Comment by stuartpb** [comment]
I have a Radeon R9 390, and when I start this game up, it's either a completely-black screen, or flickering every other frame - sometimes it's flickering to black, sometimes it's flickering to garbage VRAM data.

**Comment by smirgol** [comment]
I also have issues. First thing is that sometimes when the game starts, the screen is just black. Then at other times, I can see the game but it is flickering. The worst problem is, that when it is not black and not flickering and I can start a game, after a couple of minutes it will freeze, render garbage to one of my other monitors and just crash my X (or wayland). 
I have tried several proton and the latest GE versions. All the same. Switching around usually fixes the black screen / flickering, but they all crash my X session after a couple of minutes.

I have tried with disabling vulkan by using `PROTON_USE_WINED3D=1`, but somehow MangoHud still says its using Vulkan. I then figured the game uses a Vulkan renderer and, according to the discussions in the steam forums, there seems to be issues with the games renderer. So not sure if anything can be done about these crashes within proton.

[sysinfo](https://gist.github.com/smirgol/982717473afb360b59a4d600b36d5a4f)
[Proton log](https://gist.github.com/smirgol/98123273e60ed2a59dd63ce68fe43fb1)

**Comment by hakzsam** [comment]
@smirgol Are you still able to reproduce on your Navi?

**Comment by smirgol** [comment]
Unfortunately yes. All the problems mentioned still exist. It either renders a black screen, or it flickers and if it does none of both, which is rare and can only be achieved with switching the prefix back and forth between different versions, it will crash my X when loading a save or after playing for a very few minutes.

**Comment by hakzsam** [comment]
Can you attach the output of `vulkaninfo`?

**Comment by smirgol** [comment]
Here it is:
[vulkaninfo.txt](https://github.com/ValveSoftware/Proton/files/7703739/vulkaninfo.txt)

**Comment by smirgol** [comment]
With Proton 6.3-8 it finally seems to run pretty stable for me. No flickering, no crashes.

**Comment by brandonegbert** [comment]
I'm still seeing the previously described behavior.  Main menu flickers, GPU crashes ingame.  Have to restart X to get the session going again.

[vulkaninfo.txt](https://github.com/ValveSoftware/Proton/files/7902715/vulkaninfo.txt)
[steam_sys_info.txt](https://github.com/ValveSoftware/Proton/files/7902716/steam_sys_info.txt)
[journalctl.txt](https://github.com/ValveSoftware/Proton/files/7902717/journalctl.txt)
[steam-760650.log](https://github.com/ValveSoftware/Proton/files/7902723/steam-760650.log)

Probably the interesting bit:
```
amdgpu: The CS has been cancelled because the context is lost.
00:08:26.499684|RENDER         | 682:transfer.cpp                   Vulkan 1.1 graphics API system call failed.
Update your video card drivers at:
https://www.nvidia.com/download/index.aspx
or
https://www.amd.com/en/support

( vkQueueSubmit( _transfer_queue, 1, &submit_info, nullptr ) )

Return code: VK_ERROR_DEVICE_LOST
X connection to :99.0 broken (explicit kill or server shutdown).
```

**Comment by hakzsam** [comment]
@brandonegbert Are you still able to reproduce your problem with up-to-date Mesa?

**Comment by brandonegbert** [comment]
Yes.
[steam-760650.log](https://github.com/ValveSoftware/Proton/files/8799866/steam-760650.log)
[vulkaninfo.txt](https://github.com/ValveSoftware/Proton/files/8799872/vulkaninfo.txt)
System Info: https://gist.github.com/brandonegbert/9aa15a320b4f62fce21620d4ad8c3ec6


```
Video Card:
    Driver:  AMD AMD Radeon RX 5700 XT (navi10, LLVM 13.0.1, DRM 3.46, 5.18.0-zen1-1-zen)
    Driver Version:  4.6 (Compatibility Profile) Mesa 22.1.0
```

**Comment by hakzsam** [comment]
I can actually reproduce this easily, there is a TON of Vulkan validation errors which means this is likely a game bug.

**Comment by hakzsam** [comment]
Does RADV_DEBUG=zerovram help?

**Comment by brandonegbert** [comment]
Yes, got past the loading screen with that and didn't have any obvious corruption.  Only played for a minute so not sure if there would be errors in longer session.  I'll try a longer session in the next day or two to confirm.

[steam-760650.log](https://github.com/ValveSoftware/Proton/files/8802222/steam-760650.log)

Looks like that's already used as a default for vkd3d as a workaround, any downsides to using it?

**Comment by hakzsam** [comment]
RADV_DEBUG=zerovram shouldn't be necessary because in Vulkan, VRAM allocations aren't supposed to the zeroed but in practice a bunch of games rely on this... Good to know, let me know, if it works fine I will add a work in RADV for this game.

**Comment by brandonegbert** [comment]
Yep, played for a couple hours and it seemed fine the entire time.

**Comment by hakzsam** [comment]
No GPU hangs and no flickering now?

**Comment by brandonegbert** [comment]
Yep, same experience as it was on Windows.  No flickering, GPU hangs, or visible artifacting.  Performance seemed about as I would expect as well.

**Comment by hakzsam** [comment]
I merged this workaround in RADV today https://gitlab.freedesktop.org/mesa/mesa/-/commit/fe9f1d064af05464e5a31c74ea0d063b8c20611d

**Comment by kisak-valve** [comment]
Tracking note: Dropping the Mesa / RADV because the driver workaround has landed in mesa 22.1.2.
